spec/models/search_spec.rb

```diff
@@ -0,0 +1,90 @@
+require "rails_helper"
+
+RSpec.describe Search do
+  describe "#search_laa_reference" do
+    let(:search) { described_class.search(query).pluck(:id) }
+    let(:prepare) { build(:submission).tap(&:save) }
+
+    before { prepare }
+
+    context "when search text is LAA reference" do
+      let(:query) { "LAA-123456" }
+
+      it "returns the record" do
+        expect(search).to eq([prepare.id])
+      end
+    end
+
+    context "when search text is LAA reference does not match any records" do
+      let(:query) { "LAA-121212" }
+
+      it "returns the record" do
+        expect(search).to be_empty
+      end
+    end
+
+    context "when search text is full number part of LAA reference" do
+      let(:query) { "123456" }
+
+      it "returns the record" do
+        expect(search).to be_empty
+      end
+    end
+
+    context "when search text is partial number part of LAA reference" do
+      let(:query) { "345" }
+
+      it "returns the record" do
+        expect(search).to be_empty
+      end
+    end
+
+    context "when search text is 'laa'" do
+      let(:query) { "laa" }
+
+      it "returns the record" do
+        expect(search).to be_empty
+      end
+    end
+
+    context "when search text is full defendant name" do
+      let(:query) { "joe bloggs" }
+
+      context "with with prior authority application" do
+        let(:prepare) { build(:submission, :with_pa_version).tap(&:save) }
+
+        it "returns the record" do
+          expect(search).to eq([prepare.id])
+        end
+      end
+
+      context "with non-standard magistrate application" do
+        let(:prepare) { build(:submission, :with_nsm_version).tap(&:save) }
+
+        it "returns the record" do
+          expect(search).to eq([prepare.id])
+        end
+      end
+    end
+
```

Comment:
what do we expect to happen on a partial match??

i.e. 
```
let(:query) { "bob bloggs" }
```

should that match this record?

@jogilder

Reply:
addition would we be expecting 
```
let(:query) { "job blog" }
```
to match??